### PR TITLE
Update tutorial and introductory mobilito text

### PR DESCRIPTION
**Pull request by JeffAbrahamson**
Update the tutorial text and flow to be more natural and easier to read/ingest.

Part of #969.

### CODE REVIEW

File: transport_nantes/mobilito/views.py [hunk 1]
```diff
@@ -29,6 +29,111 @@
...
+        """Return the set of pages the user has seen.
+
+        The set is based on the tutorial cookie, and so expires when
+        it does.
+
+        """
+        if 'user' in request and request.user.is_authenticated:
+            mobilito_user = get_MobilitoUser(request)
+            if mobilito_user.completed_tutorial_timestamp:
+                return all_tutorial_pages
```
Comment (Shriukan33) [inline transport_nantes/mobilito/views.py:91]: self.all_tutorial_pages
all_tutorial_pages was defined class level and should be accessed through self. 

In this context, evaluating all_tutorial_pages would raise a NameError

File: transport_nantes/mobilito/views.py [hunk 2]
```diff
@@ -221,19 +224,17 @@ def get_context_data(self, **kwargs) -> dict:
...
-        # because it is the first time), we need to redirect to the tutorial.
-        if self.get_pages_to_visit():
+        tutorial_state = TutorialState()
+        if tutorial_state.pages_to_see(self.request):
+            next_pag = tutorial_state.next_page_to_see(request)
             return HttpResponseRedirect(
                 reverse('mobilito:tutorial',
-                        kwargs={'tutorial_page': self.get_next_tutorial_page()}))
-        return self.render_to_response(context)
+                        kwargs={'tutorial_page': next_page}))
```
Comment (Shriukan33) [inline transport_nantes/mobilito/views.py:246]: L233 : next_pag
L236 : next_page

You'd throw

File: transport_nantes/mobilito/views.py [hunk 3]
```diff
@@ -29,6 +29,111 @@
...
+
+        """
+        if 'user' in request and request.user.is_authenticated:
+            mobilito_user = get_MobilitoUser(request)
+            if mobilito_user.completed_tutorial_timestamp:
+                # User has seen all tutorial pages, nothing left to see.
+                return set()
+        return self.all_tutorial_pages - self.pages_seen(request) - set([this_page])
+
+    def next_page_to_see(self, request, this_page) -> str:
```
Comment (Shriukan33) [inline transport_nantes/mobilito/views.py:115]: The return type hinting is missing the "None", you could use Union[str, None] to tell that the function may return None as well, not only strings

File: transport_nantes/mobilito/views.py [hunk 4]
```diff
@@ -279,16 +282,9 @@ def post(self, request: HttpRequest, *args, **kwargs) \
...
-        except Exception as e:
-            logger.error(
-                f"Can't send email, user={request.user.email}, {e}")
-        else:
-            request.session.pop("location", None)
-            request.session.pop("mobilito_session_id", None)
-        return super().get(request, *args, **kwargs)
+#    def get_context_data(self, **kwargs) -> dict:
+#        context = super(TemplateView, self).get_context_data(**kwargs)
+#        return context
```
Comment (Shriukan33) [inline transport_nantes/mobilito/views.py:297]: Why commenting out instead of deleting it ?

File: transport_nantes/mobilito/views.py [hunk 5]
```diff
@@ -391,13 +387,18 @@ def prepare_email(
     return email
 
 
-def get_MobilitoUser(self: Union[TemplateView, FormView]) -> Union[MobilitoUser, None]:
-    """Get the MobilitoUser object for the current user"""
-    if self.request.user.is_authenticated:
+def get_MobilitoUser(request):
```
Comment (Shriukan33) [inline transport_nantes/mobilito/views.py:400]: Could have left the return type hint `Union[MobilitoUser, None]` :slightly_smiling_face:

**Review by Shriukan33 — CHANGES_REQUESTED** [review]
I do have a TypeError at /mobilito/formulaire-adresse

next_page_to_see() missing 1 required positional argument: 'this_page'

Also I don't understand your commit message in 435a713380bc14f8690168857d3712436cf76b0c

File: transport_nantes/mobilito/tests.py [hunk 6]
```diff
@@ -1,10 +1,38 @@
...
+
+
+class TutorialViewTests(TestCase):
+
+    def setUp(self):
+        self.tutorial_state = TutorialState()
+
+    def test_state_progression(self):
+        client = Client()
+        response_1 = client.get("/mobilito/tutoriel/presentation/")
```
[the remark this answers is not in the document]
Comment (JeffAbrahamson) [reply transport_nantes/mobilito/tests.py:31]: good point, my bad!

File: transport_nantes/mobilito/views.py [hunk 7]
```diff
@@ -29,6 +29,111 @@
...
 
+class TutorialState:
+    """Track tutorial pages viewed.
+
+    We track tutorial page views with a short-timeout cookie that
+    represents a set of tutorial pages seen.  If the cookie is absent
+    (and the user is not authenticated with a record of having
+    completed the tutorial), then the set of pages seen is empty.
+    Once the user has seen all the pages, and only then, we permit
+    recording sessions.
```
[the remark this answers is not in the document]
Comment (JeffAbrahamson) [reply transport_nantes/mobilito/views.py:40]: For another commit, but agreed.

File: transport_nantes/mobilito/views.py [hunk 8]
```diff
@@ -29,6 +29,111 @@
...
+        """
+        if 'user' in request and request.user.is_authenticated:
+            mobilito_user = get_MobilitoUser(request)
+            if mobilito_user.completed_tutorial_timestamp:
+                return all_tutorial_pages
+        tutorial_cookie = request.COOKIES.get(
+            self.tutorial_cookie_name, '')
+        if not tutorial_cookie:
+            return set()
+        pages_seen: set = pickle.loads(b64decode(bytes.fromhex(tutorial_cookie)))
```
[the remark this answers is not in the document]
Comment (JeffAbrahamson) [reply transport_nantes/mobilito/views.py:96]: This code is no longer present.

File: transport_nantes/mobilito/views.py (hunk 5 above)
Comment (JeffAbrahamson) [reply transport_nantes/mobilito/views.py:400]: Sure, I started to refactor that function, then put it back but not quite all the way clearly.

File: transport_nantes/mobilito/views.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (JeffAbrahamson) [reply transport_nantes/mobilito/views.py:96]: Oh, I didn't realise that pickle runs a byte code machine!  That does change the scenario.

**Review by Shriukan33 — APPROVED** [review]

**Comment by Shriukan33** [comment]
LGTM

**Pull request merged**